### PR TITLE
ADOP-2859 update citzen create application class

### PR DESCRIPTION
**Pull request by hfilmore**
This is @NewmanJustice 's change. Creating a new PR because I accidentally merged and reverted https://github.com/hmcts/adoption-cos-api/pull/1093

- [ ] [SonarCloud](https://sonarcloud.io/project/pull_requests_list?id=uk.gov.hmcts.reform%3Aadoption-cos-api) has been reviewed

### JIRA link (if applicable) ###
[ADOP-2859](https://tools.hmcts.net/jira/browse/ADOP-2859)

##### What’s changed ##### 
- Updated src/test/java/uk/gov/hmcts/reform/adoption/adoptioncase/event/CitizenCreateApplicationTest.java.
- Added a new parameterized test (shouldFormatHyphenatedCaseRef) using @CsvSource.
- Added coverage for:
16-digit case IDs (standard path)
shorter IDs that require left zero-padding
edge numeric inputs (1 and 0)
Assertions verify formatted output remains in ####-####-####-#### structure via the existing aboutToSubmit callback path.
##### Why this change ##### 
- Strengthens confidence that hyphenatedCaseRef is consistently generated for both typical and edge-case IDs.
- Reduces regression risk in callback output consumed downstream.
##### Impact ##### 
Test update to reflect changes in CitizenCreateApplication
##### Validation ##### 
- New parameterized unit test added to existing suite in CitizenCreateApplicationTest.
 -Existing tests in the class remain intact and continue to cover DSS metadata and base case setup.
##### What changed #####
- Updated src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/event/CitizenCreateApplication.java:
- Added defensive validation in aboutToSubmit(...) to reject invalid input:
details == null
details.getData() == null
details.getId() == null
details.getId() <= 0
- Throws IllegalArgumentException when required case details are missing/invalid.
- Extracted case reference formatting into formatHyphenatedCaseRef(Long caseId) and used it when setting hyphenatedCaseRef.

##### Why this change ##### 
- Prevents callback execution with invalid/null case payloads, reducing risk of null-handling failures at runtime.
- Ensures case reference formatting remains consistent (####-####-####-####) across typical and edge-case IDs.
##### Impact ##### 
- Behavioural improvement in callback robustness for invalid inputs.
- No API contract changes.
- Test coverage increased for case reference formatting logic and edge cases.
##### Validation ##### 
- Unit tests in CitizenCreateApplicationTest cover expected formatting scenarios.
- Existing assertions for DSS metadata and default adoption type remain in place. 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/event/CitizenCreateApplication.java [hunk 1]
```diff
@@ -43,25 +38,23 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(final CaseDet
...
+        // setDssMetaData(data);
+        // CHECKSTYLE:ON
         return AboutToStartOrSubmitResponse.<CaseData, State>builder()
             .data(data)
             .build();
     }
-
+    // CHECKSTYLE:OFF
+    /*
     private void setDssMetaData(CaseData data) {
```
Comment (hfilmore) [inline src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/event/CitizenCreateApplication.java:58]: Are you planning to remove this commented out code if it passes testing?  If not I need to raise a ticket on the backlog to do it after 6 months or something.

File: src/test/java/uk/gov/hmcts/reform/adoption/adoptioncase/event/CitizenCreateApplicationTest.java [hunk 2]
```diff
@@ -21,6 +22,7 @@ class CitizenCreateApplicationTest extends EventTest {
     @InjectMocks
     private CitizenCreateApplication citizenCreateApplication;
 
+    @Disabled
     @Test
     @DisplayName("Testing submitted event for citizen case creation with dss meta data")
```
Comment (hfilmore) [inline src/test/java/uk/gov/hmcts/reform/adoption/adoptioncase/event/CitizenCreateApplicationTest.java:27]: you've commented out this code, how is this test passing?!

File: src/test/java/uk/gov/hmcts/reform/adoption/adoptioncase/event/CitizenCreateApplicationTest.java [hunk 3]
```diff
@@ -39,10 +41,64 @@ void testingCitizenSubmissionWith_dssDataAboutToSubmit() {
         assertThat(callbackData.getDssHeaderDetails()).isEqualTo("Child Details");
     }
 
+    @Test
+    @DisplayName("Testing submitted event for citizen case creation with null case details")
+    void testingCitizenSubmissionWithNullCaseDetails() {
+        var exception = org.junit.jupiter.api.Assertions.assertThrows(IllegalArgumentException.class, () -> {
```
Comment (hfilmore) [inline src/test/java/uk/gov/hmcts/reform/adoption/adoptioncase/event/CitizenCreateApplicationTest.java:47]: sort out the import

File: src/functionalTest/java/uk/gov/hmcts/reform/adoption/citizen/ApplicationCreateFT.java
Comment (hfilmore) [inline src/functionalTest/java/uk/gov/hmcts/reform/adoption/citizen/ApplicationCreateFT.java:47]: I don't understand how tests asserting the dss data is present are passing when you've commented out that code...

File: src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/event/CitizenCreateApplication.java (hunk 1 above)
Comment (NewmanJustice) [reply src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/event/CitizenCreateApplication.java]: Done — re-enabled `setDssMetaData(data)` and removed the temporary commented-out DSS block/checkstyle suppression.

File: src/test/java/uk/gov/hmcts/reform/adoption/adoptioncase/event/CitizenCreateApplicationTest.java (hunk 2 above)
Comment (NewmanJustice) [reply src/test/java/uk/gov/hmcts/reform/adoption/adoptioncase/event/CitizenCreateApplicationTest.java]: Fixed — removed `@Disabled` from the DSS metadata unit test now that DSS metadata population is active again.

File: src/test/java/uk/gov/hmcts/reform/adoption/adoptioncase/event/CitizenCreateApplicationTest.java (hunk 3 above)
Comment (NewmanJustice) [reply src/test/java/uk/gov/hmcts/reform/adoption/adoptioncase/event/CitizenCreateApplicationTest.java]: Sorted — switched to static `assertThrows` import and updated usages.

File: src/functionalTest/java/uk/gov/hmcts/reform/adoption/citizen/ApplicationCreateFT.java
Comment (NewmanJustice) [reply src/functionalTest/java/uk/gov/hmcts/reform/adoption/citizen/ApplicationCreateFT.java]: Aligned now — DSS metadata is populated again in `CitizenCreateApplication`, so these FT assertions are consistent with the callback output.

**Comment by NewmanJustice** [comment]
**Two follow-ups from the DSS review round**

**1. `aboutToSubmit` now returns `errors` rather than throwing.** The guard previously threw `IllegalArgumentException`. From a CCD callback that becomes an HTTP 500, which data-store discards the body of and retries 3× (`CallbackService` `@Retryable`, `maxAttempts = 3`) before surfacing a 502 — the citizen gets a dead end with no actionable message. Returning the errors list instead yields a 422 carrying `callbackErrors`, which ExUI renders as a proper validation summary with the form still usable. This also matches `CitizenSubmitApplication` and the other error sites in the service, and the callback shape in `ARCHITECTURE.md`. Worth noting this event is configured `.retries(120, 120)`, so a throw was being treated as a transient fault — wrong for permanent input validation.

**2. The `"0, 0000-0000-0000-0000"` row was removed from the `shouldFormatHyphenatedCaseRef` `CsvSource`.** It can no longer pass now the guard rejects `id <= 0`; that case is covered by `testingCitizenSubmissionWithIdIsZero` instead, alongside null-details, null-data, no-id and negative-id.

One caveat for transparency: the guard is defensive rather than a fix for an observed defect. `SubmitCaseTransaction` assigns the case reference *before* invoking the callback, and CCD serialises `id` from that reference, so `details`/`data`/`id` should always be populated in production. Happy to drop the guard entirely if you'd prefer the smaller diff — the other 28 callbacks in this service don't null-check `details`.

Also restored `Jenkinsfile_CNP` to match master — an earlier merge on this branch had reintroduced a stale July commit that unpinned the Infrastructure library and dropped the Azure login steps, undoing #1106 and #1109. That file is no longer part of this PR's diff.

File: src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/event/CitizenCreateApplication.java (hunk 1 above)
Comment (hfilmore) [reply src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/event/CitizenCreateApplication.java]: Sorry Steve, the plan was to remove this altogether, not to remove the comment!

**Review by chak-shing-lo-justice — DISMISSED** [review]
lgmt

**Review by chak-shing-lo-justice — APPROVED** [review]
